vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "mmap_file_allocator_factory.h"
+#include "mmap_file_allocator.h"
+#include <vespa/vespalib/stllike/asciistream.h>
+
+namespace vespalib::alloc {
+
+MmapFileAllocatorFactory::MmapFileAllocatorFactory()
+    : _dir_name(),
+      _generation(0)
+{
+}
+
+MmapFileAllocatorFactory::~MmapFileAllocatorFactory()
+{
+}
+
+void
+MmapFileAllocatorFactory::setup(const vespalib::string& dir_name)
+{
+    _dir_name = dir_name;
```

Comment:
Do we need locking here?

Reply:
It should not be needed as long as the method is not misused.